server/py/services/api/crud/tags.py

```diff
@@ -84,7 +92,11 @@ def delete_tag_from_objects(
         delete_func = kind_to_function_names.get(tag_objects.kind, {}).get("delete")
         if not delete_func:
             raise mlrun.errors.MLRunNotFoundError(
-                f"couldn't find delete function for object kind: {tag_objects.kind}"
+                f"Couldn't find delete function for object kind: {tag_objects.kind}"
+            )
+        if tag == "latest":
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Cannot delete 'latest' tag from objects kind: {tag_objects.kind}"
```

Comment:
move out to a function (DRY)
```suggestion
        if tag == "latest":
            raise mlrun.errors.MLRunInvalidArgumentError(
                f"`latest` is a preserved tag name and cannot be deleted / mutated"
```